compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -1436,6 +1436,13 @@ impl<'a> Resolver<'a> {
                                         self.interner.function_visibility(id),
                                     );
                                 }
+
+                                if let Some(func_meta) = self.interner.func_meta.get(&id) {
+                                    self.interner.add_reference(
+                                        (DependencyId::Function(id), func_meta.location),
+                                        (DependencyId::FunctionCall, hir_ident.location),
+                                    );
+                                }
```

Comment:
Doing that will produce more code after walking the graph and collecting relations. I was hoping to avoid that by having simple resolution mechanism.
Perhaps graph describing Node types and edges describing type of relationship would suit us better?